lib/ansible/modules/windows/win_cert_stat.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2016, Ansible, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_cert_stat
+version_added: "2.10"
+short_description: Get information from the Windows Certificate Store
+description:
+- Returns information about a certificate in the Windows Certificate Store.
+options:
+  thumbprint:
+    description:
+    - The thumbprint as a hex string of the certificate to find.
+    - See M(win_certificate_store) for how to format the thumbprint.
+    type: str
+    required: yes
+  store_name:
+    description:
+    - The store name to use when searching for a certificate.
+    type: str
+    default: My
+  store_location:
+    description:
+    - The store location to use when searching for a certificate.
+    type: str
+    choices: [ CurrentUser, LocalMachine ]
+    default: LocalMachine
+seealso:
+- module: win_stat
+- module: win_reg_stat
+- module: win_certificate_store
+author:
+- Micah Hunsberger (@mhunsber)
+'''
+
+EXAMPLES = r'''
+- name: Obtain information about a certificate in the computer's personal store
+  win_cert_stat:
+    thumbprint: BD7AF104CF1872BDB518D95C9534EA941665FD27
+  register: mycert
+
+- name: Obtain information about a certificate in the root store
+  win_cert_stat:
+    store_name: Root
+    thumbprint: A43489159A520F0D93D032CCAF37E7FE20A8B419
+  register: ca
+
+# Import a pfx and then get information on the certificates
+- name: Import pfx certificate that is password protected
+  win_certificate_store:
+    path: C:\Temp\cert.pfx
+    state: present
+    password: VeryStrongPasswordHere!
+  become: yes
+  become_method: runas
+  register: mycert
+
+- name: Obtain information on each certificate that was touched
+  win_cert_stat:
+    thumbprint: "{{ item }}"
+  register: mycert_stats
+  loop: "{{ mycert.thumbprints }}"
+'''
+
+RETURN = r'''
+archived:
```

Comment:
If we wanted to go the route of an `_info` module instead we should return a list of certs that contain these entries.

Reply:
How would I note in the return documentation yml that it is a list type with complex elements?

Reply:
Have a look at [ec2_instances_info](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_instance_info.py#L75) for how to document a list of complex elements.